### PR TITLE
Merge `series/0.22` into `series/0.23`

### PR DESCRIPTION
**Pull request by danicheg**

### CODE REVIEW

File: client/shared/src/main/scala/org/http4s/client/package.scala [hunk 1]
```diff
@@ -29,6 +29,7 @@ package object client extends ClientTypes {
 trait ClientTypes {
   import org.http4s.client._
 
+  @deprecated("Is a Blaze detail.  Will be removed from public API.", "0.22.9")
```
Comment (danicheg) [inline client/shared/src/main/scala/org/http4s/client/package.scala:32]: _Reminder_: we need to delete this when merging `series/0.23` into `main`.

**Review by rossabaker — APPROVED** [review]
:+1: if green

**Comment by danicheg** [comment]
Uh oh, within `series/0.23` have the 'green CI' isn't so easy 😞

**Comment by rossabaker** [comment]
Yeah, I am trying to fix one more thing in Blaze for $WORK, and then I want to focus on a release train, but CI is in terrible shape in 0.23, so that's going to be a blocker.

**Pull request merged**